controller/DeliveryMgmt.php

```diff
@@ -0,0 +1,157 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA;
+ */
+
+namespace oat\taoLtiConsumer\controller;
+
+use oat\generis\model\kernel\persistence\smoothsql\search\ComplexSearchService;
+use oat\generis\model\OntologyAwareTrait;
+use oat\generis\model\OntologyRdfs;
+use oat\taoLtiConsumer\model\delivery\factory\LtiDeliveryFactory;
+use oat\taoLtiConsumer\model\delivery\form\LtiWizardForm;
+use oat\tao\model\taskQueue\TaskLogActionTrait;
+use oat\taoDeliveryRdf\model\DeliveryFactory;
+use oat\taoDeliveryRdf\model\tasks\CompileDelivery;
+use oat\taoDeliveryRdf\view\form\WizardForm;
+use oat\taoDeliveryRdf\model\NoTestsException;
+use oat\taoLti\models\classes\ProviderService;
+use oat\taoLtiConsumer\model\delivery\form\NoLtiProviderException;
+
+/**
+ * Controller to managed assembled deliveries
+ *
+ * @author CRP Henri Tudor - TAO Team - {@link http://www.tao.lu}
+ * @package taoDelivery
+ */
+class DeliveryMgmt extends \tao_actions_RdfController
+{
+    use OntologyAwareTrait;
+    use TaskLogActionTrait;
+
+    /**
+     * Manage the delivery wizard form
+     *
+     * Render a creation form if form has not been submitted
+     * If has been validated use WizardFormFactory to enqueue the delivery creation
+     *
+     * @return mixed
+     * @throws \common_Exception
+     * @throws \common_ext_ExtensionException
+     */
+    public function wizard()
+    {
+        $this->defaultData();
+        $formOptions = [
+            'class' => $this->getCurrentClass(),
+        ];
+
+        $noAvailableTest = $noAvailableProvider = false;
+
+        try {
+            try {
```

Comment:
`try` within `try` seems dangerous to me.
How about a sub-function?

Reply:
There is a big try catch to avoid unexpected 500; two sub try are here to manage at form level

Reply:
Ok. Still difficult to understand to me...
Any functional test planned?